src/controller/DevicePairingDelegate.h

```diff
@@ -109,6 +109,12 @@ class DLL_EXPORT DevicePairingDelegate
      * in order to resume the commissioning process.
      */
     virtual void OnScanNetworksFailure(CHIP_ERROR error) {}
+
+    /**
+     * @brief
+     *  Called to check if the discover-once option is enabled.
```

Comment:
Needs a description of what `discover once` means. I am unsure if discover once is the best naming as it seems to control if re-try on PASE is being performed or not.

Reply:
Maybe some error handling flags would be more appropriate, since this is how the pairer seems to be used. Something like:

```
enum class PairingErrorHandling {
   kReportError, // default
   kSkipRetry, // or some other name .. I am still somewhat unclear on the intent here
}
```

Reply:
`discover once` was introduced by https://github.com/project-chip/connectedhomeip/pull/21712 for TC-CADMIN-1.9.
It looks like a switch to make the controller only send PASE to the first discovered node.

Reply:
Note that "discover once" is really a hack we should not have; it's introducing a special codepath just for certification testing that does not actually run in production...

Reply:
So... shall I keep trying to fix a hack of a hack?

Reply:
That sort of depends on how much you care about the hack "working well"... just keeping in mind that it's not affecting any production code at the moment.